### PR TITLE
chore(e2e): More E2E tests for the Documents tab COMPASS-5469

### PR DESCRIPTION
**Pull request by lerouxb**
Leaving the ones that test the list/json/table view editors and their context buttons for a separate PR.

### CODE REVIEW

File: packages/compass-e2e-tests/helpers/commands/run-find.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import type { CompassBrowser } from '../compass-browser';
+import * as Selectors from '../selectors';
+
+export async function runFind(
```
Comment (lerouxb) [inline packages/compass-e2e-tests/helpers/commands/run-find.ts:4]: Pulled this out because we're now using it in 4 places in 2 files.

**Comment by lerouxb** [comment]
This is new. The import modal never appeared after clicking the button:
screenshot-Collection-import-supports-JSON-files

File: packages/compass-export-to-language/src/components/editor/editor.jsx [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 import PropTypes from 'prop-types';
-import jsBeautify from 'js-beautify';
 
 import 'ace-builds';
 import AceEditor from 'react-ace';
 import 'ace-builds/src-noconflict/mode-javascript';
 import 'ace-builds/src-noconflict/mode-csharp';
 import 'ace-builds/src-noconflict/mode-python';
 import 'ace-builds/src-noconflict/mode-java';
+import 'ace-builds/src-noconflict/mode-ruby';
```
Comment (lerouxb) [inline packages/compass-export-to-language/src/components/editor/editor.jsx:11]: This was missed when we added Ruby recently.

File: packages/compass-export-to-language/src/components/editor/editor.jsx [hunk 3]
```diff
@@ -17,44 +17,19 @@ import styles from './editor.module.less';
...
   }
 
   componentDidUpdate() {
-    if (!this.props.isInput) {
-      if (this.props.error) {
-        this.editor.setValue('');
-        this.editor.session.setMode('ace/mode/' + this.props.outputLang || 'javascript');
-        this.editor.clearSelection();
-      } else {
-        const output = this.props.showImports && this.props.imports !== '' ?
```
Comment (lerouxb) [inline packages/compass-export-to-language/src/components/editor/editor.jsx:47]: There were three copies of this code which I now factored out. Also a bug where copy to clipboard didn't take the showImports and imports props into account.

**Comment by Anemy** [comment]
Is that image supposed to be a gif in your comment on the import modal button fix? Maybe I'm not seeing what you're referring to.

**Review by Anemy — APPROVED** [review]
lgtm - watching the e2e tests fly is mad fun.
Clicking around the export to language really makes me want to rework the ux on the checkbox - I feel we shouldn't show that `Include Import Statements` checkbox on `Node` and `Python`.

**Comment by lerouxb** [comment]
@Anemy 

> Is the image supposed to be a gif in your comment on the import modal button fix? Maybe I'm not seeing what you're referring to - the copy button including the imports? nice fix.

The image is just a test flake I got once ever. A test clicked the "Add Data" button then clicked "Import File" and then nothing happened. Haven't gotten it since and as a result I haven't fixed it. My guess is something wasn't initialised yet and just ignored the event.

File: packages/compass-export-to-language/src/components/editor/editor.jsx (hunk 3 above)
Comment (lerouxb) [reply packages/compass-export-to-language/src/components/editor/editor.jsx:47]: see https://jira.mongodb.org/browse/COMPASS-4572

**Pull request merged**